data_steward/test/unit_test/achilles_heel_test.py

```diff
@@ -27,81 +30,105 @@ def setUpClass(cls):
 
     def setUp(self):
         self.hpo_bucket = gcs_utils.get_hpo_bucket(test_util.FAKE_HPO_ID)
+        self.dataset = bq_utils.get_dataset_id()
         test_util.empty_bucket(self.hpo_bucket)
-        test_util.delete_all_tables(bq_utils.get_dataset_id())
+        test_util.delete_all_tables(self.dataset)
 
-    def _load_dataset(self):
+    def _load_dataset(self, hpo_id):
         for cdm_table in resources.CDM_TABLES:
             cdm_file_name = os.path.join(test_util.FIVE_PERSONS_PATH, cdm_table + '.csv')
             if os.path.exists(cdm_file_name):
                 test_util.write_cloud_file(self.hpo_bucket, cdm_file_name)
             else:
                 test_util.write_cloud_str(self.hpo_bucket, cdm_table + '.csv', 'dummy\n')
-            bq_utils.load_cdm_csv(FAKE_HPO_ID, cdm_table)
-
-    def test_heel_analyses(self):
+            bq_utils.load_cdm_csv(hpo_id, cdm_table)
+
+        # ensure concept table exists
+        if not bq_utils.table_exists(common.CONCEPT):
+            bq_utils.create_standard_table(common.CONCEPT, common.CONCEPT)
+            q = """INSERT INTO {dataset}.concept
+            SELECT * FROM {vocab}.concept""".format(dataset=self.dataset, vocab=common.VOCABULARY_DATASET)
+            bq_utils.query(q)
+
+    @staticmethod
+    def get_mock_hpo_bucket():
+        bucket_env = 'BUCKET_NAME_' + FAKE_HPO_ID.upper()
+        hpo_bucket_name = os.getenv(bucket_env)
+        if hpo_bucket_name is None:
+            raise EnvironmentError()
+        return hpo_bucket_name
+
+    @mock.patch('gcs_utils.get_hpo_bucket')
+    def test_heel_analyses(self, mock_hpo_bucket):
         # Long-running test
-        self._load_dataset()
-
-        # populate achilles first
+        mock_hpo_bucket.return_value = self.get_mock_hpo_bucket()
         test_util.get_synpuf_results_files()
-        test_util.populate_achilles(self.hpo_bucket, include_heel=False)
-
-        achilles_heel.create_tables(FAKE_HPO_ID, True)
-        achilles_heel.run_heel(hpo_id=FAKE_HPO_ID)
-        cmd = sql_wrangle.qualify_tables(
-            'SELECT COUNT(1) FROM %sachilles_heel_results' % sql_wrangle.PREFIX_PLACEHOLDER, FAKE_HPO_ID)
-        result = bq_utils.query(cmd)
-        self.assertEqual(ACHILLES_HEEL_RESULTS_COUNT, int(result['rows'][0]['f'][0]['v']))
-        cmd = sql_wrangle.qualify_tables(
-            'SELECT COUNT(1) FROM %sachilles_results_derived' % sql_wrangle.PREFIX_PLACEHOLDER, FAKE_HPO_ID)
-        result = bq_utils.query(cmd)
-        self.assertEqual(ACHILLES_RESULTS_DERIVED_COUNT, int(result['rows'][0]['f'][0]['v']))
+
+        # create randomized tables to bypass BQ rate limits
```

Comment:
There is no TTL but they are deleted during the `tearDown()` stage of the test.

Reply:
Got it, thanks. I existing issue, but this does introduce some problems when two of these tests run simultaneously since all tests will delete all tables. Something also worth fixing in in the future.

Reply:
Agree this is an existing issue. Currently, we encounter this when two commits are made to the same branch within a short while, resulting in two circle tests on the same branch.